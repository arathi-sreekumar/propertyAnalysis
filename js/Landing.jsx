import React from 'react';
import AveragePropertyPrice from './components/AveragePropertyPrice';

const Landing = () => (
	<div className="landing">
		<input type="text" placeholder="Search" />
		<a> or Browse All</a>
		<br />
		<AveragePropertyPrice />
	</div>
);

export default Landing;
