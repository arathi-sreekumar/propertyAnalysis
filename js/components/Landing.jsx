// @flow

import React from 'react';
import { connect } from 'react-redux';
import Search from './Search/Search';

const Landing = (props: {}) => (
	<div>
		<h1 className="title">Property Information</h1>
		<Search {...props} />
	</div>
);

const mapStateToProps = state => ({
	searchTerm: state.searchTerm
});

export default connect(mapStateToProps)(Landing);
