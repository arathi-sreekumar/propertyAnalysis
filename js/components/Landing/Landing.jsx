// @flow

import React from 'react';
import Search from '../Search/Search';

const Landing = () => <Search />;

export default Landing;
