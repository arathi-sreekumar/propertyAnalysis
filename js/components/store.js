// @flow

import { createStore, compose } from 'redux';
import rootReducer from './reducers';

// Redux store to keep all redux parameters
const store = createStore(
	rootReducer,

	/* Adding redux Devtool support */
	compose(
		typeof window === 'object' && typeof window.devToolsExtension !== 'undefined'
			? window.devToolsExtension()
			: f => f
	)
);

export default store;
