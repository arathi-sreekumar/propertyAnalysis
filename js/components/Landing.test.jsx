import React from 'react';
import { render } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import store from './store';
import Landing from './Landing';

describe('Landing', () => {
	it('renders correctly', () => {
		const component = render(
			<Provider store={store}>
				<MemoryRouter>
					<Landing />
				</MemoryRouter>
			</Provider>
		);
		expect(component).toMatchSnapshot();
	});
});
