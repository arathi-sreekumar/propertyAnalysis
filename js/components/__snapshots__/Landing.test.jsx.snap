// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing renders correctly 1`] = `
<div>
  <h1
    class="title"
  >
    Property Information
  </h1>
  <form
    class="search field has-addons"
  >
    <div
      class="control is-expanded"
    >
      <input
        class="input is-primary"
        placeholder="Search"
        type="text"
        value=""
      />
    </div>
    <div
      class="control"
    >
      <button
        class="button is-primary"
        type="submit"
      >
        Search
      </button>
    </div>
  </form>
</div>
`;
